allauth/account/views.py

```diff
@@ -290,6 +290,13 @@ def get_context_data(self, **kwargs):
         )
         return ret
 
+    def get_initial(self):
+        if self.request.GET.get("email", None):
+            initial = super().get_initial()
+            initial["email"] = self.request.GET.get("email")
+            return initial
+        return super().get_initial()
```

Comment:
Would also be nice to make this a bit more DRY. You're invoking super().get_initial twice. How about:

```
initial = super().get_initial():
if ...:
  prefill
return initial
```

Reply:
noted